packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/ImageAndTextViewer/ImageAndTextViewer.spec.js

```diff
@@ -174,6 +175,22 @@ describe('ImageAndTextViewer', function () {
       expect(screen.getByRole('button', { name: 'SlideTutorial.StepNavigation.next' })).to.exist()
     })
 
+    it('should call the onReady prop', function () {
```

Comment:
I tried to add a similar test for an image location first, but I couldn't figure out with [the `useSubjectImage` hook and other related loading code](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleImageViewer/SingleImageViewerContainer.js#L40-L53).

Reply:
If your mock image URL returns 404, then `onError` will probably fire rather than `onReady`. 

I’m not sure if JSDOM supports image load events.

Reply:
I tried variations on the mock subject image location, including a nonsense url, url that would error, and url that I thought would succeed, and none worked as I expected with `onReady` or `onError` to be called or not called. I think I'm missing something fundamental, maybe there's a completely different way to structure the tests to accomplish desired result, or maybe as you state JSDOM doesn't support image load events. I'll revisit (we'll have to when refactoring SingleImageViewer with RTL?), but will merge this for now to fix the bug.

Reply:
This is for an older version of the events library, but [explicitly firing a load event](https://stackoverflow.com/questions/60481331/testing-img-onload-img-onerror-using-jest-and-react-testing-library) on the image might work: `fireEvent.load(image)`. The syntax probably needs to be updated for version 14 of the user events library.

Reply:
I'd also guess that the image in the tests doesn't have any layout, since we aren't rendering it in a browser, so no `clientHeight`, `clientWidth` etc.